content/en/blog/2025-12-12-flux-aio-kubernetes-mtls-and-the-chicken-and-egg-problem/index.md

```
@@ -0,0 +1,132 @@
+    name: cozystack-assets-tls
+```
+
+And now Flux can download all the necessary assets.
+
+In my opinion, this is a beautiful hack worth sharing, as it teaches us something new. However, I wouldn't recommend considering this idea as a best practice.
+
+I think we'll get rid of it in the future. We're gradually transitioning to `source-watcher` and the ability to store artifacts directly in `OCIRepository`. This way, Flux will download and collect all necessary artifacts directly from the specified OCI image or Git repository.
+
+You can view (and review) the full code of the PR here:
+
+- https://github.com/cozystack/cozystack/pull/1698
```

markdownlint-cli2 (0.18.1)

Bare URL used

(MD034, no-bare-urls)